api/v1alpha1/watcher_types.go

```diff
@@ -52,18 +51,8 @@ const (
 
 // GatewayConfig is used to select an Istio Gateway object in the cluster.
 type GatewayConfig struct {
-	// NamespacedName takes precedence over LabelSelector if configured. Format to use: "namespaceName/gatewayName"
-	// TODO: Add validation: <name><slash><namespace>
-	// +kubebuilder:validation:Optional
-	// +kubebuilder:validation:MinLength=3
-	// +kubebuilder:validation:MaxLength=127
-	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?[/][a-z]([-a-z0-9]*[a-z0-9])?$`
-	NamespacedName string `json:"namespacedName,omitempty"`
```

Comment:
any particular reason why we exclude direct configuration for the gateway? It's certainly less error-prone than configuration by label selector - so I'd say it's safer.

Reply:
IMO, on one hand, it's valid concern to allow multiple options to choose gateway, but on the other hand, it's not necessary to define and offering such lose contract if we can restrict them. Take this as an example, by allowing multiple way to choose gateway, then certainly we need to concern the priories for those options, in the original implementation, `NamespacedName` will overwrite selector, which confusing end user if they defined a selector but find out it's not working. This confusing also apples to error-prone, basically if the `NamespacedName` configured wrong, then the selector is useless, we will return error.

The previous concern to allow selecting by name is mainly because gateway is not managed by us in the beginning, so in case the permission problem (we can't add labels to the gateway), to choose by name make sense. But since now the gateway are fully controlled by us, it's not necessary to offer so much options. 


We can choose any of them of cause, but use label selector to select resource is more natural way, that's the reason drop the `NamespacedName`.

Reply:
As for the "API simplicity", i.e: "keep the API minimal" - you are right.
We could resolve all other problems, but - at this point - it makes no sense to introduce sophisticated configuration for simple things.
:+1: